server/modules/surveyRdb/repository/dataView/read.js

```diff
@@ -35,7 +38,17 @@ const _getSelectFields = ({ viewDataNodeDef, columnNodeDefs, nodeDefCols, editMo
       viewDataNodeDef.columnRecordUuid,
       viewDataNodeDef.columnUuid,
       // selected node def columns
-      ...nodeDefCols.map((nodeDefCol) => new ColumnNodeDef(viewDataNodeDef, nodeDefCol).namesFull).flat(),
+      ...nodeDefCols
+        .map((nodeDefCol) => {
+          const columnNodeDef = new ColumnNodeDef(viewDataNodeDef, nodeDefCol)
+
```

Comment:
no need for _isBooleanAndHasLabels method. a variable `isBooleanAndHasLabels` is enough.